packages/core/src/operation.ts

```diff
@@ -40,6 +42,10 @@ export interface OperationFunction<TOut> extends Labelled {
   (task: Task<TOut>): Operation<TOut>;
 }
 
+export interface OperationObject<TOut> extends Labelled {
```

Comment:
As mentioned in chat, I think that this should not be a `Labelled`.

```suggestion
export interface OperationObject<TOut> {
```

Reply:
In that case we have to add checks to almost all places where we are trying to get labels from operation. Because `name` and `labels` fields are optional, I guess is better to make it extends from `Labelled` to reduce change impact

Reply:
I disagree. Even though it is painful for us, the user never works with the labels directly, they use `withLabels()` and `setLabels()` so I think that we need to cope with the pain internally to simplify the external interface because having a `name` property is very common for an object, and we should avoid that conflict even if that name is also a string.